[New product] Node-feature-discovery

Pyronewbic opened the pull request.

Pyronewbic committed:
Added Node-feature-discovery

captn3m0 committed:
[nfd] Minor changes

captn3m0 commented on products/node-feature-discovery.md:
```diff
@@ -0,0 +1,60 @@
+  nfd-gc -version
+  nfd-master -version
+  nfd-topology-updater -version
+  nfd-worker -version
```
Are these likely to report different versions? I'd prefer if we could provide a single command.

Alternatively, since the version is same as on the docker image, would a kubectl command be better here ([list the image across all namespaces (with a grep)](https://kubernetes.io/docs/tasks/access-application-cluster/list-all-running-container-images/#list-all-container-images-in-all-namespaces)). That would work on non-helm installs as well.

Pyronewbic replied:
These all report the same version - but I'll be swapping it with the kubectl list images command instead.

captn3m0 commented on products/node-feature-discovery.md:
```diff
@@ -0,0 +1,60 @@
+  nfd-master -version
+  nfd-topology-updater -version
+  nfd-worker -version
+releasePolicyLink: https://github.com/kubernetes-sigs/node-feature-discovery/releases
```
This doesn't provide any policy information. The [Security Policy](https://github.com/kubernetes-sigs/node-feature-discovery?tab=security-ov-file#readme) also doesn't mention any clear support policy, only k8s-compatibility policy, which isn't enough for here.

Do we have a better link?

Pyronewbic replied:
Unfortunately, I haven't been able to find anything about this.

captn3m0 commented on products/node-feature-discovery.md:
```diff
@@ -0,0 +1,60 @@
+> [node-feature-discovery](https://kubernetes-sigs.github.io/node-feature-discovery/) is a a Kubernetes add-on for detecting hardware
+> features and system configuration. It advertises these features using node labels and optionally node extended resources, annotations and node taints.
+
+Each release is churned out at a period of nearly two weeks, and the latest two releases are considered supported.
```
Needs a citation, and if this is an estimate, we should make it clear:
```suggestion
Looking at prior releases, a new release is made every two weeks, and the latest two releases
receive bug fixes.
```

captn3m0 reviewed (changes requested):
few questions.

Pyronewbic committed:
Update products/node-feature-discovery.md

Co-authored-by: Nemo <commits@captnemo.in>

Pyronewbic committed:
Update node-feature-discovery.md

Pyronewbic committed:
Update node-feature-discovery.md

captn3m0 commented on products/node-feature-discovery.md:
```diff
@@ -0,0 +1,61 @@
+  kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec['initContainers', 'containers'][*].image}" |\
+  tr -s '[[:space:]]' '\n' |\
+  sort |\
+  uniq -c
```
Wouldn't this show images for all containers instead of just for node-feature-discovery?

Pyronewbic replied:
I've updated it to exec into a deployment and print the version out.

Pyronewbic committed:
Update node-feature-discovery.md

Updated Product with way of getting version (work for k8s deploy & hem deploys)

Pyronewbic committed:
Update node-feature-discovery.md

captn3m0 reviewed (approved):
LGTM.
